CRM/Utils/Mail/EmailProcessor.php

```diff
@@ -117,13 +117,15 @@ private static function _process($civiMail, $dao, $is_create_activities) {
 
           // if its the activities that needs to be processed ..
           try {
-            $mailParams = CRM_Utils_Mail_Incoming::parseMailingObject($mail, $createContact, FALSE, $emailFields);
+            $mailParams = CRM_Utils_Mail_Incoming::parseMailingObject($mail, $createContact, [$incomingMail->getFrom()], $emailFields);
```

Comment:
this is a fugly way to pass it in - but this function doesn't have long-to-live in general.... or not as it currently appears